File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       // Arrays.sort(nums);
        int i = 0, j = nums.length-1;

        HashMap<Integer, Integer>hm = new HashMap<>();
        for(int l = 0; l < nums.length; ++l){
            i = nums[l];
            j = target-i;
            if(hm.containsKey(j)==true)return new int[]{hm.get(j),l};
            else{hm.put(i,l);}
        }

        return new int[2];
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int n = nums.size();
        int i = 0, j = n - 1;

        while( i <= j ){
            int mid = i + ( j - i )/2;

            int num = nums[mid];
            if( nums[ (mid-1+n) % n ] > num && nums[ (mid+1+n) % n ] > num)return num;
            else if(num > nums[j])i = mid + 1;
            else{
                j = mid - 1;
            }
        }

        return nums[i];
    }
};
